lib/Conversion/HWArithToHW/HWArithToHW.cpp

```diff
@@ -476,11 +521,19 @@ class HWArithToHWPass : public HWArithToHWBase<HWArithToHWPass> {
 
     // Generic conversion and legalization patterns for operations that we
```

Comment:
Not for this PR, but having to list out all the ops from all of the dialects which may use ui/si is not ideal. Is there some way to deal with this generically?

Reply:
As i see it, this would require:
1. ui/si usage is defined through an op interface
2. The `OpConversionPattern`s of this PR are then replaced with `OpInterfaceConversionPattern`s to target this interface.